### DISCUSSION TITLE
Voltage sensing with stepdown

### DISCUSSION THREAD

**Discussion by frietim** [OpenI6X/opentx#274] [General]
Hi, I'm powering my radio off a 2s lipo via a 6v bec and I'm wondering if it is known where the voltage sensing happens on the pcb. If So, could it be possible to disconnect the voltage sensor from the power input and connect it to the input side of the stepdown to measure the real battery voltage?

**Comment by psicho100pro** [comment]
Hi, I was wondering this too, the MCU has a voltage detection pin, but according to the datasheet I found that its maximum does not exceed the maximum supply voltage. Maybe I was looking wrong, but I'd probably be afraid to risk permanent damage over it. I didn't bother with it anymore. But there are people who know more, maybe they have a solution. Sort of like a voltage divider. 1/2 and v set correction in settings.

**Comment by ajjjjjjjj** [comment]
Hi, it's PC.0 pin number. It's located in the top row of the MCU by looking at it in the PCB orientation.

**Reply by frietim** [reply]
It's a bit late but I'm looking into this again. I didn't find a pin labeled PC.0, were you talking about pcb labeling or STM32 pin labeling?

**Reply by ajjjjjjjj** [reply]
Standard STM32F072 pin number, it's not exposed as a pad on pcb.

**Comment by frietim** [comment]
![Bild1-min](https://user-images.githubusercontent.com/111421916/200171686-f9b0a461-7f59-46b7-ae78-f80dcaba3ea0.png)
found it, it seems to be connected to a trace on the backside, however it actually seems to be broken out to a "Battery" pad on the right, it's measuring about 2.2V on 6.3V displayed input voltage, sadly I cant trace it back any further, its connected to R23 but I don't know where that goes to.

**Reply by ajjjjjjjj** [reply]
That R23 is probably part of voltage divider, that's why you read 2.2V on 6.3V input.

**Reply by Muhaiminul-Hasan** [reply]
hi, that's what I found out using the continuity mode on my multimeter, hope that helps.

![200171686-f9b0a461-7f59-46b7-ae78-f80dcaba3ea0](https://user-images.githubusercontent.com/50764001/213167112-7fa9ea76-f0c4-41b0-8085-308df3dcb9e8.jpg)

**Reply by psicho100pro** [reply]
If we could verify that the "battery pad" is really only connected to pin 0, then we could certainly come up with something. But since I haven't tried to figure out what the structure actually serves here, other than powering the LCD backlight, then it's a guess. I have two of these radios, but I don't want to do much digging.

**Reply by Ultraexpert** [reply]
Maybe an email to flysky ?

**Reply by Muhaiminul-Hasan** [reply]
Hi, I actually did it, just cut or disconnect carefully at the black line as the image below (Be aware of the PCB 'via' right in between the resistors, keep it connected to the resistor marked 121 for the Backlight LED ), and connect your Battery (+ve) terminal to the resistor marked 01C.
In this method, I didn't mess with the voltage divider or the pad marked Battery. I just connected the battery to the existing connection circuitry as it should, except bypassing the power. It Works, but do it at your own risk.
Happy Modding!

Before:
![200171686-f9b0a461-7f59-46b7-ae78-f80dcaba3ea0](https://user-images.githubusercontent.com/50764001/213167112-7fa9ea76-f0c4-41b0-8085-308df3dcb9e8.jpg)
After:
![200171686-f9b0a461-7f59-46b7-ae78-f80dcaba3ea0](https://user-images.githubusercontent.com/50764001/213168031-be5c127f-311b-4d71-94f1-46950682d538.jpg)


My Radio:
![IMG20230111004114](https://user-images.githubusercontent.com/50764001/212939875-553a57f6-714a-40bf-8328-9f9038519ad8.jpg)

PS: My connection looks different, but it's the same.
I actually cut at point A initially and connected the battery (+ve) to point X, but that is also connected to the Backlight LED. So, if you use a 2S battery, that might be a problem, so I used a jumper wire for the LED instead.
![200171686-f9b0a461-7f59-46b7-ae78-f80dcaba3ea0](https://user-images.githubusercontent.com/50764001/212960129-840f0c46-ad67-4964-a681-f2ab7a0a63a2.jpg)

**Reply by psicho100pro** [reply]
The simplest solution then can be this https://lampatronics.com/product/bta06-sensitive-gate-triac/
I was thinking about something else, -> edit
I add, something like that https://i.imgur.com/vNu6ln2.png

**Reply by Muhaiminul-Hasan** [reply]
It works, Just connect your BEC to the Battery connector normally to power the radio. The red (B+) is the voltage sense, and connect your Battery (+ve) terminal to the resistor marked 01C after disconnecting at the black line. You don't even need to calibrate voltage. You may need to adjust the Battery low alarm according to your battery voltage. You don't need anything else. Thanks.
Before:
![200171686-f9b0a461-7f59-46b7-ae78-f80dcaba3ea0](https://user-images.githubusercontent.com/50764001/213167112-7fa9ea76-f0c4-41b0-8085-308df3dcb9e8.jpg)
After:
![200171686-f9b0a461-7f59-46b7-ae78-f80dcaba3ea0](https://user-images.githubusercontent.com/50764001/213168031-be5c127f-311b-4d71-94f1-46950682d538.jpg)

**Reply by psicho100pro** [reply]
This is what the datasheet says:
VDD–VSS External main supply voltage - 0.3 - 4.0 V
VDDIO2 –V SS External I/O supply voltage - 0.3 - 4.0 V
VDDA –VSS External analog supply voltage - 0.3 - 4.0 V
VDD –V DDA Allowed voltage difference for VDD > VDDA - - 0.4 V
VBAT–V SS External backup supply voltage - 0.3 - 4.0 V
Input voltage on FT and FTf pins V SS - 0.3 VDDIOx + 4.0 V
Input voltage on TTa pins V SS - 0.3 - 4.0 V
BOOT0 0 - 9.0 V
Input voltage on any other pin V SS - 0.3 -  4.0 V
|∆VDDx | Variations between different VDD power pins - 50 mV
|VSSx - VSS| Variations between all the different ground pins - 50 mV

**Reply by psicho100pro** [reply]
@Muhaiminul-Hasan Can you post a photo of the 2s battery here and how many volts can be seen on the LCD?  According to the photos below, you are using a 4.4V battery, and for that this modification is completely meaningless.

**Reply by Muhaiminul-Hasan** [reply]
Are you worried about the board frying up?

R21 (01C) = 10k resistor
R22 (512) = 5k resistor
R21 and R22 create a 1/3 voltage divider. 
fully charged 2S lipo = 8.4v
the voltage on the BATTERY pad should be <2.8v also R23 is 200 ohm resistor for current limiting.
I power mine with a 5v boost converter, before the mod it used to put out 1.66v, one-third of 5v.
I think 2S lipo will work just fine. But you can add a 5k resistor in series if you want and calibrate it in the menu.

**Reply by Muhaiminul-Hasan** [reply]
> @Muhaiminul-Hasan Can you post a photo of the 2s battery here and how many volts can be seen on the LCD? According to the photos below, you are using a 4.4V battery, and for that this modification is completely meaningless.

I did this modification because It used to put out a constant 5v regardless of the battery voltage as it's connected to the boost module or BEC. Now it shows the actual battery voltage.
 
I don't power it from the 4.4v battery directly, it goes through the 5v boost module so that the battery voltage can go lower than the operating voltage of the TX but still power the board just fine and perform the same even on a low battery. Otherwise, the LCD may flicker and can turn off when the battery is very low.
 
I think 1S lipo with a 5v boost module is enough to power the two 3.3v regulators on board (one for the TX module and the other for the rest of the board), the reason it runs on 6v out from the factory is so that it can use four AA batteries (1.5v x 4 = 6v). Powering it with a 2S battery and a linear BEC is not efficient. It is bulky and doesn't look good hanging outside the case, also charging a 2S battery is difficult too. Using 1S, you can charge it via USB and a cheap charging module.
 
This setup is the cleanest I think, Next, I'll do an external TX module mod when I get one like a TBS crossfire Nano.

Anyway, I'll test with a 2s battery and post a picture soon if I can. Thanks.

BTW similar mod has been done to the receiver too, Ex: FlySky FS-iA6B, FS-iA10B
even a 3S battery is tested.
![33376871-b02c3cdc-d50f-11e7-8224-11dde7d21e70](https://user-images.githubusercontent.com/50764001/213272418-a24bc4b0-f8df-4d94-87df-2077d3906d4c.jpg)
![Sensor voltage vs battery voltage](https://user-images.githubusercontent.com/50764001/213275140-7799e3c4-7a8e-4a4f-bf0e-79aa8d8c14d4.jpg)
![Screenshot 2023-01-19 012206](https://user-images.githubusercontent.com/50764001/213275170-72a965cf-80bb-4dba-aa82-5af0072e014f.jpg)

**Comment by Muhaiminul-Hasan** [comment]
I want to implement that too, I am using a 5v dc-dc boost module, so it makes it difficult to know the battery capacity, because it always outputs 5v. I probed the "BATTERY" pad on the board and it puts out 1.66v, "BATT_IN" takes 5.00 v in my case, hope someone figures this out.
![IMG202211301](https://user-images.githubusercontent.com/50764001/213171031-a54c45ba-bb59-4faa-952f-b0b2d8cbff33.jpg)
![IMG20221130182949](https://user-images.githubusercontent.com/50764001/213171071-c2b644be-4033-4a5d-814a-2ee7728e4923.jpg)
![IMG20221130183504](https://user-images.githubusercontent.com/50764001/213171075-2ddcadd6-ec39-4e89-a9d6-cc3f0f35a1b2.jpg)
![IMG20230111003635](https://user-images.githubusercontent.com/50764001/213172987-43ecb77d-5a4c-4cba-92f3-e4a2909d09bf.jpg)
![IMG20230111004010](https://user-images.githubusercontent.com/50764001/213173028-4fe4dea3-29b2-475e-ac7a-8c2ad82879b9.jpg)
